airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalClient.java

```diff
@@ -347,7 +347,7 @@ public ManualSyncSubmissionResult startNewCancelation(final UUID connectionId) {
             Optional.of("Didn't manage cancel a sync for: " + connectionId),
             Optional.empty());
       }
-    } while (connectionManagerWorkflow.getState().isRunning());
+    } while (isWorkflowStateRunning(getConnectionManagerName(connectionId)));
```

Comment:
This is the key line that should fix the acceptance test flakiness. We now simply terminate the while loop if a workflow becomes unreachable